src/net_processing.cpp

```diff
@@ -2658,8 +2657,15 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // set nodes not capable of serving the complete blockchain history as "limited nodes"
         pfrom.m_limited_node = (!(nServices & NODE_NETWORK) && (nServices & NODE_NETWORK_LIMITED));
 
-        {
+        // We don't initialize the m_tx_relay data structure if:
+        // - this is an outbound block-relay-only connection; or
+        // - fRelay=false and we're not offering NODE_BLOOM to this peer
+        //   (NODE_BLOOM means that the peer may turn on tx relay later)
+        const bool disable_tx_relay = pfrom.IsBlockOnlyConn() ||
```

Comment:
Inverted this could be:
`const bool enable_tx_relay = !pfrom.IsBlockOnlyConn() && (fRelay || (pfrom.GetLocalServices() & NODE_BLOOM))`.

Don't you think it's better?